src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -539,8 +539,15 @@ public override int Read (byte [] buffer, int offset, int count)
 					lock (dataLock) {
 						// this is the same object, it was done to make the cleanup
 						data.Dequeue ();
-						current?.Dispose ();
 						currentStream?.Dispose ();
+						// We cannot use current?.Dispose. The reason is the following one:
+						// In the DidReceiveResponse, if iOS realizes that a buffer can be reused,
+						// because the data is the same, it will do so. Such a situation does happen
+						// between requests, that is, request A and request B will get the same NSData
+						// (buffer) in the delegate. In this case, we cannot dispose the NSData because
+						// it might be that a different request received it and it is present in
+						// its NSUrlSessionDataTaskStream stream. We can only trust the gc to do the job
+						// which is better than copying the data over. 
```

Comment:
Don't you need to remove this as well?

https://github.com/xamarin/xamarin-macios/blob/1b3672e2e358fdb90228aaf267b38159f9f8fac0/src/Foundation/NSUrlSessionHandler.cs#L468-L469

Reply:
Indeed, the whole `Dispose` override can be removed

Reply:
Ture, it happen less often, but will be an issue too.